src/main/java/slimeknights/tconstruct/items/GadgetItems.java

```diff
@@ -17,61 +17,42 @@
+import slimeknights.tconstruct.library.registration.object.EnumObject;
+import slimeknights.tconstruct.library.registration.object.ItemObject;
 import slimeknights.tconstruct.shared.block.SlimeBlock;
 
-import static slimeknights.tconstruct.common.TinkerPulse.injected;
-
-@ObjectHolder(TConstruct.modID)
 @Mod.EventBusSubscriber(modid = TConstruct.modID, bus = Mod.EventBusSubscriber.Bus.MOD)
 public final class GadgetItems {
 
-  public static final Item stone_stick = injected();
-
-  public static final SlimeSlingItem slime_sling_blue = injected();
-  public static final SlimeSlingItem slime_sling_purple = injected();
-  public static final SlimeSlingItem slime_sling_magma = injected();
-  public static final SlimeSlingItem slime_sling_green = injected();
-  public static final SlimeSlingItem slime_sling_blood = injected();
-
-  public static final SlimeBootsItem slime_boots_blue = injected();
-  public static final SlimeBootsItem slime_boots_purple = injected();
-  public static final SlimeBootsItem slime_boots_magma = injected();
-  public static final SlimeBootsItem slime_boots_green = injected();
-  public static final SlimeBootsItem slime_boots_blood = injected();
-
-  public static final PiggyBackPackItem piggy_backpack = injected();
-
-  public static final FancyItemFrameItem jewel_item_frame = injected();
-  public static final FancyItemFrameItem aluminum_brass_item_frame = injected();
-  public static final FancyItemFrameItem cobalt_item_frame = injected();
-  public static final FancyItemFrameItem ardite_item_frame = injected();
-  public static final FancyItemFrameItem manyullyn_item_frame = injected();
-  public static final FancyItemFrameItem gold_item_frame = injected();
-  public static final FancyItemFrameItem clear_item_frame = injected();
-
-  public static final GlowBallItem glow_ball = injected();
-  public static final EflnBallItem efln_ball = injected();
+  private static final Item.Properties gadgetProps = new Item.Properties().group(TinkerRegistry.tabGadgets);
+  private static final ItemDeferredRegister ITEMS = new ItemDeferredRegister(TConstruct.modID);
 
-  @SubscribeEvent
-  static void registerItems(final RegistryEvent.Register<Item> event) {
-    BaseRegistryAdapter<Item> registry = new BaseRegistryAdapter<>(event.getRegistry());
+  public static void init() {
+    IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
 
-    registry.register(new Item((new Item.Properties()).group(TinkerRegistry.tabGadgets)), "stone_stick");
-
-    for (SlimeBlock.SlimeType type : SlimeBlock.SlimeType.VISIBLE_COLORS) {
-      registry.register(new SlimeSlingItem(), "slime_sling_" + type.getName());
-      registry.register(new SlimeBootsItem(type), "slime_boots_" + type.getName());
+    ITEMS.register(modEventBus);
+    for (SlimeBlock.SlimeType type : SlimeBlock.SlimeType.values()) {
+      System.out.println(type);
+      System.out.println(type.toString());
```

Comment:
Leftover debug?

Reply:
yep. part of why the thing wasn't working with remaping; SlimeType's toString
is not overriden for lowercase.